csv/src/main/java/org/motechproject/nms/csv/utils/CsvInstanceImporter.java

```diff
@@ -17,14 +30,124 @@ public CsvInstanceImporter(Class<T> type) {
 
     public T read() throws IOException {
         if (isOpen()) {
-            return getCsvReader().read(type, getFieldNames(), getProcessors());
+            T instance = createInstance();
+            Map<String, String> cells = csvReader.read(columnNames);
+            if (null != cells) {
```

Comment:
oh, and if you're going to do something on both sides of a comparison, it makes the code more readable to test for equality than to test for non equality, so the following `if` construct would me more readable:
```
if (cells == null) {
    return null;
} else {
    ....
}
```